mailpoet/lib/Segments/DynamicSegments/FilterDataMapper.php

```diff
@@ -89,6 +92,39 @@ private function getSegmentType(array $data): string {
     return $data['segmentType'];
   }
 
+  private function createAutomations(array $data): DynamicSegmentFilterData {
```

Comment:
The method contains a bunch of conditions, so it might be useful to have it covered by a test. I believe this will make it easier for our future selves to feel more confident when making changes here.
We already have a [unit test](https://github.com/mailpoet/mailpoet/blob/974cd3295065e4c50855c985d50384aa07acfa23/mailpoet/tests/unit/Segments/DynamicSegments/FilterDataMapperTest.php) you can extend.

Reply:
Thanks for pointing this out! I've updated the tests in 8964fb6aa2290a83f70ae34d995f195e2df65cfe. Please let me know if you think I missed any important cases.